yt/geometry/particle_deposit_vector.pyx

```diff
@@ -0,0 +1,195 @@
+# distutils: include_dirs = LIB_DIR
+# distutils: libraries = STD_LIBS
+"""
+Particle Deposition onto Cells
+
+
+
+
+"""
+
+
+cimport numpy as np
+
+import numpy as np
+
+cimport cython
+from libc.math cimport sqrt
+
+from yt.utilities.lib.fp_utils cimport *
+
+from .oct_container cimport Oct, OctInfo, OctreeContainer
+
+from yt.utilities.lib.misc_utilities import OnceIndirect
+
+
+cdef append_axes(np.ndarray arr, int naxes):
+    if arr.ndim == naxes:
+        return arr
+    # Avoid copies
+    arr2 = arr.view()
+    arr2.shape = arr2.shape + (1,) * (naxes - arr2.ndim)
+    return arr2
+
+cdef class ParticleDepositOperation:
```

Comment:
Could it superclass the one defined in `particle_deposit.pyx` to minimize code duplication?